model-optimizer/extensions/middle/LeakyReluPattern.py

```diff
@@ -13,21 +13,87 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
+import logging as log
 from extensions.middle.fusings import Fusing
 from extensions.middle.pass_separator import PostMiddleStart
-from mo.graph.graph import Graph
-from mo.middle.passes.leaky_relu import convert_mul_eltwise_to_leaky_relu
+from extensions.ops.activation_ops import LeakyReLU
+from mo.graph.graph import Graph, rename_nodes
 from mo.middle.replacement import MiddleReplacementPattern
 
 
-class LeakyReLU(MiddleReplacementPattern):
+class LeakyReLUFusion(MiddleReplacementPattern):
+    """
+    The transformation  finds next subgraph:
+
+    -->Data-------->Maximum-->Data
+          `-->Mul---`
+
+    and replaces with ReLU with negative slope (LeakyRelu)
+    """
     enabled = True
+    force_clean_up = True
 
     def run_after(self):
         return [Fusing]
 
     def run_before(self):
         return [PostMiddleStart]
 
-    def find_and_replace_pattern(self, graph: Graph):
-        convert_mul_eltwise_to_leaky_relu(graph)
+    @staticmethod
+    def pattern():
+        return dict(
+            nodes=[
+                ('data', dict(kind='data')),
+                ('mul_data', dict(kind='data')),
+                ('max_op', dict(kind='op', type='Maximum')),
+                ('const_op', dict(kind='op', type='Const')),
+                ('const_data', dict(kind='data')),
+                ('mul_op', dict(kind='op', type='Multiply')),
+            ],
+            edges=[
+                ('data', 'mul_op'),
+                ('mul_op', 'mul_data'),
+                ('data', 'max_op'),
+                ('mul_data', 'max_op'),
+                ('const_op', 'const_data'),
+                ('const_data', 'mul_op')
+            ],
+        )
+
+    def replace_pattern(self, graph: Graph, match: dict):
+        mul_node = match['mul_op']
+        const_node = match['const_op']
+        max_node = match['max_op']
+        max_name = max_node.soft_get('name', max_node.id)
+
+        # Check that nodes satisfies conversion requirements
+        connected_ports = [port for port in max_node.in_ports().values() if not port.disconnected()]
+        if len(connected_ports) > 2:
+            log.debug('Maximum layer ({}) can not participate in conversion to leaky ReLU due to it has more than two '
+                      'inputs ({})'.format(max_node.id, len(connected_ports)))
+            return
+
+        const_value = const_node.out_port(0).data.get_value()
+        if const_value is None or const_value.size != 1:
+            log.debug('Mul layer ({}) can not participate in conversion to leaky ReLU due to value {}'
+                      ''.format(mul_node.id, const_value))
+            return
+
+        scalar_value = const_value.item(0)
+        if len(mul_node.out_port(0).get_destinations()) > 1:
```

Comment:
we can allow such cases, we will have LeakyReLU and separate path with Mul and it seems to me the code below operates the way I described it.

Reply:
Updated transformation and unit test for such a case.